ekncontent/ekncontent/eknc-domain.c

```diff
@@ -830,8 +858,13 @@ on_xapian_query_response (GObject *source,
           return;
         }
 
+      /* Note that we pass another closure here instead of just the task
```

Comment:
We're already setting the task-level data to `QueryState`, whereas these callbacks are sort of per-object. What I wanted to do was create a closure for each callback to indicate which index was in use, which I don't think I would be able to do on the `QueryState` object itself since that is shared between all callback invocations.